src/test/ui/traits/issue-85735.rs

```diff
@@ -0,0 +1,13 @@
+// Regression test for the invalid suggestion in #85735 (the
+// underlying issue #21974 still exists here).
+
+#![crate_type="lib"]
```

Comment:
Nit: I guess we usually just put an empty `fn main() {}` instead of changing the crate type, but I don't think it really matters